src/types.ts

```diff
@@ -13,12 +13,16 @@ export interface LogFn {
     args: OperationArgs,
     // tslint:disable-next-line:no-any
     result: any,
-    startTime?: [number, number],
+    startTime?: HighResTime,
   ): Promise<void>;
 
-  startTimer(): [number, number];
+  startTimer(): HighResTime;
 }
 
-export type Time = number | [number, number];
+export type LevelMetadata = {level: number};
 
-export type TimerFn = (start?: [number, number]) => Time;
+export type HighResTime = [number, number]; // [seconds, nanoseconds]
+
+export type Time = number | HighResTime;
+
+export type TimerFn = (start?: HighResTime) => Time;
```

Comment:
Shouldn't `TimerFn` return `HighRestTime`, so that we don't need to cast the result of `TimerFn` [here](https://github.com/strongloop/loopback4-example-log-extension/pull/3/commits/88451e7b12c02436458fc3cf0b8e501de1de9a71#diff-930d21e839a5c8b793865de0f36206f3R37)?

Or perhaps we need a more elaborate type signature if something like that is possible in TypeScript:

```ts
export type TimerFn = () => HighResTime | (start: HighResTime) => number;
```

Maybe it's a sign that TimerFn is too complex and we should simplify it to always return the HighResTimer value?

Not a big deal. If there isn't any obvious and easy solution, then we can keep the current implementation.